components/fxa-client/android/src/main/java/mozilla/appservices/fxa/FirefoxAccount.kt

```diff
@@ -0,0 +1,158 @@
+
+    /**
+     * Fetches the profile object for the current client either from the existing cached state
+     * or from the server (requires the client to have access to the profile scope).
+     *
+     * @param ignoreCache Fetch the profile information directly from the server
+     * @return Profile representing the user's basic profile info
+     * @throws FxaException.Authentication if the account is not connected
+     */
+    abstract suspend fun getProfile(ignoreCache: Boolean = false): Profile
+
+    /**
+     * Fetches the token server endpoint, for authentication using the SAML bearer flow.
+     *
+     * @return Token server endpoint URL string
+     * @throws FxaException.Authentication if the account is not connected
+     */
+    abstract suspend fun getTokenServerEndpointURL(): String
+
+    /**
+     * Get the pairing URL to navigate to on the Authority side (typically a computer).
+     *
+     * @return The URL to show the pairing user
+     */
+    abstract fun getPairingAuthorityURL(): String
+
+    /**
+     * Returns current FxA Device ID for an authenticated account.
+     *
+     * @return Current device's FxA ID
+     * @throws FxaException.Authentication if the account is not connected
+     */
+    abstract fun getCurrentDeviceId(): String
+
+    /**
+     * Returns session token for an authenticated account.
+     *
+     * @return Current account's session token
+     * @throws FxaException.Authentication if the account is not connected
+     */
+    abstract fun getSessionToken(): String
+
+    /**
+     * Indicates if account needs to be re-authenticated via [beginAuthentication].
+     * Most common reason for an account to need re-authentication is a password change.
+     *
+     * TODO this may return a false-positive, if we're currently going through a recovery flow.
+     * Prefer to be notified of auth problems via [AccountObserver], which is reliable.
+     *
+     * @return A boolean flag indicating if account needs to be re-authenticated.
+     */
+    abstract fun accountNeedsReauth(): Boolean
+
+    /**
+     * This method should be called when a request made with an OAuth token failed with an
+     * authentication error. It will re-build cached state and perform a connectivity check.
+     *
+     * In time, fxalib will grow a similar method, at which point we'll just relay to it.
+     * See https://github.com/mozilla/application-services/issues/1263
+     *
+     * @param singleScope An oauth scope for which to check authorization state.
+     * @return An optional [Boolean] flag indicating if we're connected, or need to go through
+     * re-authentication. A null result means we were not able to determine state at this time.
+     */
+    abstract suspend fun checkAuthorizationStatus(singleScope: String): Boolean?
+
+    /**
+     * Register for account events
+     */
+    abstract suspend fun registerAccountEventHandler(handler: AccountEventHandler)
```

Comment:
What do you think about moving this to the constructor? forcing the consumer to make decision about what to do with the events... I could see the argument for allowing the consumer not to set the handlers so I could go for either 🤔

Reply:
I'm in the middle too.  Originally I thought it should be in the constructor, but it's possible that consumers might not be ready to construct their event handler yet.